### PR TITLE
feat: refactor to remove nonce from queries

### PR DESCRIPTION
**Pull request by krpeacock**
# Description

replaces disableNonce feature with useQueryNonces. Going forward, calls will use nonces, but queries and readstate calls will not. Queries and readsatate calls will use nonces if `useQueryNonces` is set to true.

Fixes # (issue)

# How Has This Been Tested?

Removes the nonceless e2e tests as they are no longer applicable

# Checklist:

- [x] My changes follow the guidelines in [CONTRIBUTING.md](https://github.com/dfinity/agent-js/blob/main/CONTRIBUTING.md).
- [x] The title of this PR complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/).
- [x] I have edited the CHANGELOG accordingly.
- [x] I have made corresponding changes to the documentation.

### CODE REVIEW

File: docs/generated/changelog.html [hunk 1]
```diff
@@ -12,6 +12,11 @@ <h1>Agent-JS Changelog</h1>
     <section>
       <h2>Version x.x.x</h2>
       <ul>
+        <li>
+          feat!: replaces disableNonce feature with useQueryNonces. Going forward, calls will use
```
Comment (ericswanson-dfinity) [inline docs/generated/changelog.html:16]:
```suggestion
          feat!: replaces disableNonce feature with useQueryNonces. Going forward, updates will use
```

File: packages/agent/src/agent/http/index.ts [hunk 2]
```diff
@@ -263,10 +262,28 @@ export class HttpAgent implements Agent {
     return hostname === '127.0.0.1' || hostname.endsWith('127.0.0.1');
   }
 
-  public addTransform(fn: HttpAgentRequestTransformFn, priority = fn.priority || 0): void {
-    // Keep the pipeline sorted at all time, by priority.
-    const i = this._pipeline.findIndex(x => (x.priority || 0) < priority);
-    this._pipeline.splice(i >= 0 ? i : this._pipeline.length, 0, Object.assign(fn, { priority }));
+  public addTransform(
+    type: 'call' | 'query',
```
Comment (ericswanson-dfinity) [inline packages/agent/src/agent/http/index.ts:266]: Maybe to check nomenclature: I thought we have:
- "update calls" and "query calls"
- "updates" and "queries"
- but not "calls" and "queries"

**Comment by ShuoWangNSL** [comment]
Thanks. LGTM!

File: packages/agent/src/agent/http/index.ts (hunk 2 above)
Comment (krpeacock) [reply packages/agent/src/agent/http/index.ts:266]: I can refactor for that

**Review by ericswanson-dfinity — APPROVED** [review]

**Pull request merged**